commit ea6a287ea9c27a1d3066bf1c2957b5526ff11923
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2019-06-21T22:50:13+00:00 (4 months ago)
    
    SWSUP-1216 Package resize from portal, CLI, and admin-ui ignoring package traits.

diff --git a/lib/machines.js b/lib/machines.js
index 72aacf292..3b82b4a9b 100644
--- a/lib/machines.js
+++ b/lib/machines.js
@@ -2136,6 +2136,12 @@ function resize(req, res, next) {
                     pkgRam, imgReq.max_ram));
     }
 
+
+    if (!jsprim.deepEqual(req.pkg.traits, req.machine.package.traits)) {
+        return next(new InvalidArgumentError(
+            'Resizing to package with differing traits not supported'));
+    }
+
     var callback = updateCallback(req, res, next),
         params = {
             uuid: req.params.machine,
diff --git a/package.json b/package.json
index 4c9a230ca..f7abc4549 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "cloudapi",
     "description": "Triton CloudAPI",
-    "version": "9.6.0",
+    "version": "9.7.1",
     "author": "Joyent (joyent.com)",
     "private": true,
     "engines": {
